Source/Core/VideoCommon/PixelShaderGen.h

```diff
@@ -167,7 +167,8 @@ struct pixel_shader_uid_data
   } stagehash[16];
 
   LightingUidData lighting;
-  BitFieldArray<0, 8, 1, bool, u8> texMtxInfo_n_projection;  // 8x1 bit
+  // Stored separately to guarantee that the texMtxInfo struct is 8 bits wide
+  BitFieldArray<0, 8, 1, u8> texMtxInfo_n_projection;
```

Comment:
I'm pretty sure this should be `BitFieldArray<0, 1, 8, u8>`, as you want 8 fields where each field is one bit, not 1 field that's 8 bits.  This might be the source of your fifoci diffs.  (Though I'd also recommend doing a full rebase just to be certain).  This also applies to VertexShaderGen.h.

Reply:
That would explain the static assert I was getting for the 16 bit sized one